hydromt/model/processes/grid.py

```diff
@@ -377,9 +377,8 @@ def grid_from_rasterdataset(
     else:
         raise ValueError(f"reproject_method should have length 1 or {len(variables)}")
     # Masking
-    if mask_name is not None:
-        if mask_name in grid_like:
-            ds_out = ds_out.raster.mask(grid_like[mask_name])
+    if mask_name is not None and mask_name in grid_like:
```

Comment:
Would this work? I don't think `grid_like` contains `None`.

```suggestion
    if mask_name in grid_like:
```